tests/pkg/notifications/operations/execute-status-report.go

```diff
@@ -0,0 +1,129 @@
+	formationIDPathParam           = "ucl-formation-id"
+	formationAssignmentIDPathParam = "ucl-assignment-id"
+)
+
+type FormationAssignmentRequestBody struct {
+	State         string          `json:"state"`
+	Configuration json.RawMessage `json:"configuration"`
+	Error         string          `json:"error"`
+}
+
+type ExecuteStatusReportOperation struct {
+	externalServicesMockMtlsSecuredURL string
+	client                             *http.Client
+	tenant                             string
+	assignmentSource                   string
+	assignmentTarget                   string
+	config                             string
+	statusCode                         int
+	asserters                          []asserters.Asserter
+}
+
+func NewExecuteStatusReportOperation() *ExecuteStatusReportOperation {
+	return &ExecuteStatusReportOperation{
+		statusCode: http.StatusOK,
+	}
+}
+
+func (o *ExecuteStatusReportOperation) WithExternalServicesMockMtlsSecuredURL(externalServicesMockMtlsSecuredURL string) *ExecuteStatusReportOperation {
+	o.externalServicesMockMtlsSecuredURL = externalServicesMockMtlsSecuredURL
+	return o
+}
+
+func (o *ExecuteStatusReportOperation) WithHTTPClient(client *http.Client) *ExecuteStatusReportOperation {
+	o.client = client
+	return o
+}
+
+func (o *ExecuteStatusReportOperation) WithFormationAssignment(source, target string) *ExecuteStatusReportOperation {
+	o.assignmentSource = source
+	o.assignmentTarget = target
+	return o
+}
+
+func (o *ExecuteStatusReportOperation) WithStatusCode(statusCode int) *ExecuteStatusReportOperation {
+	o.statusCode = statusCode
+	return o
+}
+
+func (o *ExecuteStatusReportOperation) WithTenant(tenant string) *ExecuteStatusReportOperation {
+	o.tenant = tenant
+	return o
+}
+
+func (o *ExecuteStatusReportOperation) WithAsserters(asserters ...asserters.Asserter) *ExecuteStatusReportOperation {
+	for i, _ := range asserters {
+		o.asserters = append(o.asserters, asserters[i])
+	}
+	return o
+}
+
+func (o *ExecuteStatusReportOperation) Execute(t *testing.T, ctx context.Context, gqlClient *gcli.Client) {
+	formationID := ctx.Value(context_keys.FormationIDKey).(string)
+
+	t.Logf("List formation assignments for formation with ID: %q", formationID)
+	listFormationAssignmentsReq := fixtures.FixListFormationAssignmentRequest(formationID, 100)
+	assignmentsPage := fixtures.ListFormationAssignments(t, ctx, gqlClient, o.tenant, listFormationAssignmentsReq)
+
+	formationAssignmentID := getFormationAssignmentIDBySourceAndTarget(t, assignmentsPage, o.assignmentSource, o.assignmentTarget)
+
+	t.Logf("Calling FA status API for formation assignment ID %q", formationAssignmentID)
+	faAsyncStatusAPIURL := strings.Replace(o.externalServicesMockMtlsSecuredURL, fmt.Sprintf("{%s}", formationIDPathParam), formationID, 1)
+	faAsyncStatusAPIURL = strings.Replace(faAsyncStatusAPIURL, fmt.Sprintf("{%s}", formationAssignmentIDPathParam), formationAssignmentID, 1)
+	reqBody := FormationAssignmentRequestBody{
+		State: "READY",
```

Comment:
I may be mistaken here, but since we aim to make this `ExecuteStatusReportOperation` generic, do we want to always respond with hardcoded `READY` state? In case we want to report another state, will we have to create another operation?